lib/tree.rb

```diff
@@ -16,42 +16,144 @@ def initialize
     @root = nil
   end
 
-  # Time Complexity: 
-  # Space Complexity: 
+  # Time Complexity: O(n)
+  # Space Complexity: O(n)
   def add(key, value)
```

Comment:
👍  The time/space complexity would be O(n) for an unbalanced tree and O(log n) for a balanced tree.